bot/seasons/evergreen/space.py

```diff
@@ -0,0 +1,205 @@
+        """Parse `str` into `datetime` or `int`. When invalid value, raise error."""
+        if argument.isdigit():
+            return int(argument)
+        try:
+            date = datetime.strptime(argument, "%Y-%m-%d")
+        except ValueError:
+            raise BadArgument(f"Can't convert `{argument}` to `datetime` in format `YYYY-MM-DD` or `int` in SOL.")
+        return date
+
+
+class Space(Cog):
+    """Space Cog contains commands, that show images, facts or other information about space."""
+
+    def __init__(self, bot: SeasonalBot):
+        self.bot = bot
+        self.http_session = bot.http_session
+
+    @group(name="space", invoke_without_command=True)
+    async def space(self, ctx: Context) -> None:
+        """Head command that contains commands about space."""
+        await ctx.send_help("space")
+
+    @space.command(name="apod")
+    async def apod(self, ctx: Context, date: Optional[str] = None) -> None:
+        """Get Astronomy Picture of Day from NASA API. Date is optional parameter, what formatting is YYYY-MM-DD."""
+        # Make copy of parameters
+        params = APOD_DEFAULT_PARAMS.copy()
+        # Parse date to params, when provided. Show error message when invalid formatting
+        if date:
+            try:
+                params["date"] = datetime.strptime(date, "%Y-%m-%d").date().isoformat()
+            except ValueError:
+                await ctx.send(f"Invalid date {date}. Please make sure your date is in format YYYY-MM-DD.")
+                return
+
+        # Do request to NASA API
+        result = await self.fetch_from_nasa("planetary/apod", params)
+
+        # Create embed from result
+        embed = Embed(title=f"Astronomy Picture of the Day - {result['date']}", description=result["explanation"])
+        embed.set_image(url=result["url"])
+        embed.set_footer(text="Powered by NASA API")
+
+        await ctx.send(embed=embed)
+
+    @space.command(name="nasa")
+    async def nasa(self, ctx: Context, *, search_term: Optional[str] = None) -> None:
+        """Get random NASA information/facts + image. Support `search_term` parameter for more specific search."""
+        # Create params for request, create URL and do request
+        params = {
+            "media_type": "image"
+        }
+        if search_term:
+            params["q"] = search_term
+
+        async with self.http_session.get(url=f"{NASA_IMAGES_BASE_URL}/search?{urlencode(params)}") as resp:
+            data = await resp.json()
+
+        # Check is there any items returned
+        if len(data["collection"]["items"]) == 0:
+            await ctx.send(f"Can't find any items with search term `{search_term}`.")
+            return
+
+        # Get (random) item from result, that will be shown
+        item = random.choice(data["collection"]["items"])
+
+        # Create embed and send it
+        embed = Embed(title=item["data"][0]["title"], description=item["data"][0]["description"])
```

Comment:
This embed generation code is essentially identical across every subcommand. How about a helper method to build an Embed from the provided parameters (title, description, image, footer)?